lib/plugins/aws/invokeLocal/resolvers/index.js

```diff
@@ -0,0 +1,160 @@
+const aws = require('aws-sdk');
+const fs = require('fs');
+
+const handlers = {
+    'AWS::S3::Bucket': require('./s3'),
+    'AWS::DynamoDB::Table': require('./dynamodb'),
+    'AWS::SNS::Topic': require('./sns'),
+    'AWS::SQS::Queue': require('./sqs'),
+    'AWS::Lambda::Function': require('./lambda')
+};
+
+let cfrServerless = null;
+let cfResources = null;
+let cfExports = null;
+let serviceConfig = null;
+let cache = null;
+const cacheFilePath = '.serverless/invoke.cache.json';
+
+async function getCFResources(serverless) {
+    if(cfrServerless === serverless) {
+        return cfResources;
+    }
+
+    const cf = new aws.CloudFormation(serviceConfig);
+
+    const resources = await cf.describeStackResources({
+        StackName: `${serverless.service.service}-${serverless.service.provider.stage}`
+    }).promise();
+
+    return resources.StackResources;
+}
+
+module.exports = async (name, environment, serverless, credentialEnvVars) => {
+    if(!cache) {
+        if(fs.existsSync(cacheFilePath)) {
+            try {
+                cache = JSON.parse(fs.readFileSync(cacheFilePath).toString('utf-8'));
+            } catch (err) {
+                cache = {};
+            }
+        } else {
+            cache = {};
+        }
+    }
+    const value = environment[name];
+    const accessKey = credentialEnvVars.AWS_ACCESS_KEY_ID;
+    const secret = credentialEnvVars.AWS_SECRET_ACCESS_KEY;
+    serviceConfig = {
+        region: (credentialEnvVars.AWS_REGION)? credentialEnvVars.AWS_REGION : serverless.service.provider.region,
+        credentials: (accessKey)? new AWS.Credentials(accessKey, secret) : undefined
+    };
+
+    if(typeof value != 'object') {
+        return;
+    }
+
+    const keys = Object.keys(value);
+    if(keys.length != 1) {
+        // Not sure how to handle this where an environment variable has multiple opperators
```

Comment:
This would work without cache, but when cache is involved this leads to duplication, as ImportValue and Ref resolve cache in the same way.  This also results in a lot more passing around of parameters.  I've made updates to get it closer to this.  I avoided the "else throw" due to variations on lambda references, which we probably shouldn't cause the invoke local to crash, but rather just not set the environment variable.  Let me know what you think.

Reply:
In this comment I meant that I wouldn't introduce any special route/validation for keys.length > 1.

I would agree that `action = 'Ref'` if there's a `value.Ref` (and if there's also `value.foo` ignore it)